File: Go/ExampleTests/common/pricing_string.go
package common

import "fmt"

type PricingString struct {
	TotalPrice string
}

func NewPricingStringFromSlice(v []string) PricingString {
	s := PricingString{}
	if len(v) > 0 { s.TotalPrice = v[0] }
	return s
}

func (s PricingString) String() string {
	return fmt.Sprintf("TotalPrice=%s", s.TotalPrice)
}

func (s PricingString) Equals(o PricingString) bool {
	return DNCEqual(s.TotalPrice, o.TotalPrice)
}

func EqualPricingStringSlices(a, b []PricingString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/pricing_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct PricingString {
    pub totalprice: String,
}

impl PricingString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            totalprice: v.get(0).copied().unwrap_or("").to_string(),
        }
    }
}

impl std::fmt::Display for PricingString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f,
            "TotalPrice={}",
            self.totalprice
        )
    }
}

impl PartialEq for PricingString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.totalprice, &other.totalprice)
    }
}
